kotlinx-coroutines-core/common/src/JobSupport.kt

```diff
@@ -517,8 +523,12 @@ public open class JobSupport constructor(active: Boolean) : Job, ChildJob, Paren
                 ?: InvokeOnCompletion(this, handler)
     }
 
-    private fun addLastAtomic(expect: Any, list: NodeList, node: JobNode<*>) =
-        list.addLastIf(node) { this.state === expect }
+    private fun addLastAtomic(expect: Any, list: NodeList, node: JobNode<*>): Boolean {
+        list.addLast(node)
+        if (this.state === expect) return true
```

Comment:
```suggestion
    private fun addLastAtomic(expectedState: Any, list: NodeList, node: JobNode<*>): Boolean {
        list.addLast(node)
        if (this.state === expectedState) return true
```

I think this naming is more explicit and avoids confusion with the `expect` soft keyword.